tests/test_sphinxcontrib_trio.py

```diff
@@ -136,6 +143,13 @@ def acm_wrapped():  # pragma: no cover
 
         check(acm_wrapped, "async-with")
 
+    if have_asynccontextmanager:
+        @asynccontextmanager
+        async def acm():  # pragma: no cover
+            yield_()
```

Comment:
This isn't valid -- it would need to be `await yield_()` and also be in a function decorated by `@async_generator`. Instead, maybe rely on the fact that `have_asynccontextmanager` will only be true on >=3.7 and native async generators exist on >=3.6, so you can just say `acm = asynccontextmanager(agen_native)` using the `agen_native` that was conditionally defined above.

Reply:
I don't really understand all of your comment, but I changed the test so that I only test what I added in functionality which is detecting the @asynccontextmanager decorator.

Reply:
`@asynccontextmanager` should decorate an async generator function. Your previous example was not an async generator; your current example isn't either. The version of `@asynccontextmanager` in the stdlib in 3.7+ doesn't seem to check, but (for example) the version in the `async_generator` library does check, and it seems clearer to give it what it's expecting regardless.

In theory you could write this as
```
@asynccontextmanager
@async_generator
async def acm():  # pragma: no cover
    await yield_()
```
but that depends on the `async_generator` library, which isn't a dependency of sphinxcontrib-trio. Or you could write it as
```
@asynccontextmanager
async def acm():  # pragma: no cover
    yield
```
but that will produce a SyntaxError on 3.5. My suggestion was that you write it as
```
acm = asynccontextmanager(agen_native)
```
using the `agen_native` async generator function that was conditionally defined further up (in the testing of `async for` autodetection). It's true that `agen_native` is only defined on 3.6+, but `asynccontextmanager` only exists on 3.7+, so there's no actual possibility of an error.